third_party/airflow/tests/unit/test_armada_operator.py

```diff
@@ -0,0 +1,33 @@
+from armada.operators.armada import ArmadaOperator
+import pytest
+
+get_lookout_url_test_cases = [
+    (
+        "http://localhost:8089/jobs?job_id=<job_id>",
+        "test_id",
+        "http://localhost:8089/jobs?job_id=test_id",
+    ),
+    (
+        "https://lookout.armada.domain/jobs?job_id=<job_id>",
+        "test_id",
+        "https://lookout.armada.domain/jobs?job_id=test_id",
+    ),
+    ("", "test_id", ""),
+]
+
+
+@pytest.mark.parametrize(
+    "lookout_url_template, job_id, expected_url", get_lookout_url_test_cases
+)
+def test_get_lookout_url(lookout_url_template, job_id, expected_url):
+    operator = ArmadaOperator(
```

Comment:
Hmm.  I think this should error out if it is running.  

We assume that someone is able to submit a job using the ArmadaClient and then use the JobService for polling in the Operator.  Looking at the code, I think there is a bug if you are able to pass tests with a None ArmadaClient and JobService.

I created test_airflow_operator_mock so I can test with mock grpc objects.  I would also suggest adding tests their with the lookout url.